executor/sort.go

```diff
@@ -141,37 +128,18 @@ func (e *SortExec) Next(ctx context.Context, req *chunk.Chunk) error {
 			return err
 		}
 	} else {
-		// Check whether the one partition is spilled.
-		// If the partition is in memory, use List.GetRow() to get better performance.
-		if !e.partitionList[0].AlreadySpilled() {
-			rowChunks := e.partitionList[0].GetList()
-			for !req.IsFull() && e.Idx < len(e.rowPtrs) {
-				rowPtr := e.partitionRowPtrs[0][e.Idx]
-				req.AppendRow(rowChunks.GetRow(rowPtr))
-				e.Idx++
-			}
-		} else {
-			for !req.IsFull() && e.Idx < len(e.rowPtrs) {
-				rowPtr := e.partitionRowPtrs[0][e.Idx]
-				row, err := e.partitionList[0].GetRow(rowPtr)
-				if err != nil {
-					return err
-				}
-				req.AppendRow(row)
-				e.Idx++
+		for !req.IsFull() && e.Idx < e.partitionList[0].NumRow() {
+			row, err := e.partitionList[0].GetRowByIdx(e.Idx)
```

Comment:
use too many RLocks in GetRowByIdx. 
It is better to keep the lowest callee function thread-safe.

Reply:
```
GetSortedRow(idx)
BenchmarkSortExec/(rows:300000,_orderBy:[0_1],_ndvs:_[0_0])-8         	       5	 236137816 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[0_1],_ndvs:_[1_0])-8         	       4	 306196869 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[0],_ndvs:_[1_0])-8           	      21	  53316005 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[1],_ndvs:_[1_0])-8           	       5	 234143878 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[0_1],_ndvs:_[10000_0])-8     	       4	 285625359 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[0],_ndvs:_[10000_0])-8       	       5	 214384946 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[1],_ndvs:_[10000_0])-8       	       5	 242921345 ns/op

Master
BenchmarkSortExec/(rows:300000,_orderBy:[0_1],_ndvs:_[0_0])-8         	       6	 202542442 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[0_1],_ndvs:_[1_0])-8         	       4	 259148025 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[0],_ndvs:_[1_0])-8           	      55	  20992548 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[1],_ndvs:_[1_0])-8           	       6	 191120944 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[0_1],_ndvs:_[10000_0])-8     	       5	 238929357 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[0],_ndvs:_[10000_0])-8       	       6	 179271168 ns/op
BenchmarkSortExec/(rows:300000,_orderBy:[1],_ndvs:_[10000_0])-8       	       6	 193617344 ns/op
```